pkg/cmd/secrets.go

```diff
@@ -218,9 +224,40 @@ func downloadSecrets(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	body, apiError := http.DownloadSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, format == "json")
-	if !apiError.IsNil() {
-		utils.HandleError(apiError.Unwrap(), apiError.Message)
+	fallbackPassphrase := getPassphrase(cmd, "fallback-passphrase", localConfig)
+	if fallbackPassphrase == "" {
+		utils.HandleError(errors.New("invalid fallback file passphrase"))
+	}
+
+	var body []byte
+	if format == "json" {
+		fallbackPath := ""
+		legacyFallbackPath := ""
+		if enableFallback {
+			fallbackPath, legacyFallbackPath = initFallbackDir(cmd, localConfig, exitOnWriteFailure)
+		}
+		secrets := fetchSecrets(localConfig, enableFallback, fallbackPath, legacyFallbackPath, fallbackReadonly, fallbackOnly, exitOnWriteFailure, fallbackPassphrase)
```

Comment:
Can you add some comments around here on what this function exactly does? Since the fallback arguments are passed in I feel like it is doing more than just fetching secrets?

Reply:
Adding godoc